### PR TITLE
Added support for internal kvdb and other changes

### PR DESCRIPTION
**Pull request by jitupawar22**
Fixed cassandra testsuite issues.
Added support for installation of portworx, that will needed for other frameworks.
Addes px image pull support.
Added installation of px with auth enabled.

Signed-off-by: Jitendra Pawar <jitendra.pawar@gmail.com>

### CODE REVIEW

File: frameworks/cassandra/tests/test_backup_and_restore.py [hunk 1]
```diff
@@ -91,3 +85,12 @@ def test_backup_and_restore_to_s3():
         'restore-s3',
         plan_parameters,
         config.get_foldered_node_address())
+
+@pytest.mark.sanity
+def test_bkp_restore_uninstall():
```
Comment (piyush-nimbalkar) [inline frameworks/cassandra/tests/test_backup_and_restore.py:90]: Why is this a separate test? Is it even testing that the uninstall has happened?

File: frameworks/portworx/tests/test_update_px_service.py [hunk 2]
```diff
@@ -92,9 +96,11 @@ def portworx_service(service_account):
 
         yield {**options, **{"package_name": config.PACKAGE_NAME}}
     finally:
-        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
-        sdk_install.portworx_cleanup()
+        return 0
```
Comment (piyush-nimbalkar) [inline frameworks/portworx/tests/test_update_px_service.py]: Why is this needed? Is the finally even needed?

File: frameworks/cassandra/tests/test_backup_and_restore.py (hunk 1 above)
Comment (jitupawar22) [reply frameworks/cassandra/tests/test_backup_and_restore.py:90]: Yes it is even testing that uninstall is happen. 
But we need this change in order to keep the cluster in same state on first failure of any test.
So moved the uninstall part from "Finally:" to separate test.

File: frameworks/portworx/tests/test_update_px_service.py (hunk 2 above)
Comment (jitupawar22) [reply frameworks/portworx/tests/test_update_px_service.py]: not needed, cleaned up.

**Review by piyush-nimbalkar — APPROVED** [review]
Squash the commits into one commit before you merge.
